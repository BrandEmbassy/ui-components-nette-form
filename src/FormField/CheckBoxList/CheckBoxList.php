<?php declare(strict_types = 1);

namespace BrandEmbassy\Components\NetteForm\FormField\CheckBoxList;

use BrandEmbassy\Components\Controls\Checkbox\Checkbox;
use BrandEmbassy\Components\DataAttributes;
use BrandEmbassy\Components\SelectList\SelectList;
use BrandEmbassy\Components\SelectList\SelectListItem;
use BrandEmbassy\Components\UiComponent;
use function array_map;

/**
 * @final
 */
class CheckBoxList implements UiComponent
{
    /**
     * @var CheckBoxListItem[]
     */
    private array $rowsData;

    private string $name;

    private ?DataAttributes $dataAttributes;


    /**
     * @param CheckBoxListItem[] $itemsData
     */
    public function __construct(string $name, array $itemsData, ?DataAttributes $dataAttributes = null)
    {
        $this->name = $name;
        $this->rowsData = $itemsData;
        $this->dataAttributes = $dataAttributes;
    }


    public function render(): string
    {
        $rows = array_map(
            function (CheckBoxListItem $rowData): SelectListItem {
                /** @var UiComponent[] $children */
                $children = [];

                $avatar = $rowData->getAvatar();
                if ($avatar !== null) {
                    $children[] = $rowData->getAvatar() ?? '';
                }

                $checkbox = new Checkbox(
                    $children,
                    $rowData->getLabelText(),
                    $this->name . '-' . $rowData->getId(),
                    $this->name,
                    $rowData->getId(),
                    $rowData->isSelected(),
                );

                return new SelectListItem([$checkbox]);
            },
            $this->rowsData,
        );

        return (new SelectList($rows, $this->dataAttributes))->render();
    }
}
