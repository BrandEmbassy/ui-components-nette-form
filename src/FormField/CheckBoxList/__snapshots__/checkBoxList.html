<div class="row padding-10" data-cy='GridRow'>
    <div class="padding-0 col-xs-12" data-cy='GridColumn'><h3 class="" data-cy='Heading'>checkBoxList</h3></div>
    <div class="padding-0 col-xs-12" data-cy='GridColumn'>
        <div class="select-list-654d" {{dataAttribute}}>
            <div class="Checkbox__CheckboxContent Checkbox__hasBorder"
                 onclick="document.getElementById('checkBoxListComponent&#x5B;&#x5D;-id-1').click();">
                <div class="Checkbox__Checkbox"><input type="checkbox" id="checkBoxListComponent&#x5B;&#x5D;-id-1"
                                                       value="id-1" name="checkBoxListComponent[]"
                                                       data-cy='Checkbox-checkBoxListComponent[]'/><label
                         onclick="return false;" for="checkBoxListComponent&#x5B;&#x5D;-id-1"></label></div>
                <div class="Checkbox__Label">check-label-1</div>
            </div>
            <div class="Checkbox__CheckboxContent Checkbox__hasBorder"
                 onclick="document.getElementById('checkBoxListComponent&#x5B;&#x5D;-id-2').click();">
                <div class="Checkbox__Checkbox"><input type="checkbox" id="checkBoxListComponent&#x5B;&#x5D;-id-2"
                                                       value="id-2" name="checkBoxListComponent[]"
                                                       data-cy='Checkbox-checkBoxListComponent[]'/><label
                         onclick="return false;" for="checkBoxListComponent&#x5B;&#x5D;-id-2"></label></div>
                <div class="Checkbox__Label">check-label-2</div>
            </div>
        </div>
    </div>
</div>
